libvast/vast/taxonomies.hpp

```diff
@@ -0,0 +1,50 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include "vast/fwd.hpp"
+
+#include <caf/meta/type_name.hpp>
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace vast {
+
+using concepts_t = std::unordered_map<std::string, std::vector<std::string>>;
+using models_t = std::unordered_map<std::string, std::vector<std::string>>;
+
+struct taxonomies {
```

Comment:
Why is this plural?

Reply:
For our current and short-term planned usages of this information don't make use of the name of a taxonomy itself. On the other hand, keeping all concepts in a single map simplifies the resolution algorithm. For those reasons I opted to store all taxonomies combined in one struct.

Reply:
Is your argument "simplicity"? As a user, I'd like to be able to exchange/select taxonomies, but I do not see how this is supposed to be enhanced from here. Why not just have a single taxonomy throughout the code for now, and then add more cross-taxonomy logic later on?

Reply:
> Why not just have a single taxonomy throughout the code for now, and then add more cross-taxonomy logic later on?

I implemented the feature according to its documentation. If I would follow your suggestion the end result would not work the way the user expects.

> As a user, I'd like to be able to exchange/select taxonomies, but I do not see how this is supposed to be enhanced from here.

This sounds like a useful feature but I would change it slightly to read: I want to set a default taxonomy, so that I don't have to prefix every concept/module name in an expression with it.